internal/provider/resource_remote_file.go

```diff
@@ -74,6 +90,18 @@ func resourceRemoteFileCreate(ctx context.Context, d *schema.ResourceData, meta
 		if err != nil {
 			return diag.Errorf("error while changing permissions of remote file with sudo: %s", err.Error())
 		}
+		if group != "" {
+			err = client.ChgrpFileSudo(path, group)
+			if err != nil {
+				return diag.Errorf("error while changing group of remote file with sudo: %s", err.Error())
+			}
+		}
+		if owner != "" {
+			err = client.ChownFileSudo(path, owner)
+			if err != nil {
+				return diag.Errorf("error while changing owner of remote file with sudo: %s", err.Error())
+			}
+		}
 	} else {
```

Comment:
Would like to have ownership functionality for non-`sudo` connections as well. Does not seem like the scp client supports ownership, so I think we should do it with `chown` and `chgrp`. Could you modify `ChgrpFileSudo` and `ChownFileSudo` to take a `sudo: bool` parameter, and only include `sudo` in the command when `sudo` is `true`? We can then reuse the functions for non-`sudo` connections.